shaders/water.gdshader

```diff
@@ -84,7 +85,22 @@ vec3 get_view_position_from_uv(vec2 uv, float depth, mat4 inv_proj_m)
 	return view_position.xyz /= view_position.w;
 }
 
-vec3 get_ssr_color(vec3 surface_view_position, vec3 normal_view_space, vec3 view_view_space, mat4 proj_m, mat4 inv_proj_m)
+float get_screen_border_alpha(vec2 screen_position)
+{
+    vec2 shifted_screen_position = 4.0 * screen_position * (1.0 - screen_position);
+	float mask = shifted_screen_position.x * shifted_screen_position.y; // ranging from 0.0 at the edges to 1.0 in the center 
+
+	// An offset in the [0.0, 0.5] range for ssr_screen_border_fadeout values > 0.75
+	// which is subtracted from the result of smoothstep.
+	// This ensure alpha smoothly transitions to zero when ssr_screen_border_fadeout is approaching 1.0.	
+	float offset = mix(0.0, 0.5, (clamp(ssr_screen_border_fadeout, 0.75, 1.0)-0.75) / 0.25);
+ 
+	float alpha = smoothstep(0.0, 2.0 * ssr_screen_border_fadeout, mask) - offset;
```

Comment:
```suggestion
	float alpha = clamp(smoothstep(0.0, 2.0 * ssr_screen_border_fadeout, mask) - offset, 0.0, 1.0);
```

here is a clamp missing. Otherwise negative alphas can happen (starts with fadeout 0.8+):
![Godot_v4 2 2-stable_win64_gyvmeb8xgD](https://github.com/marcelb/GodotSSRWater/assets/308374/f8b779f8-5fc1-4c10-a6d5-c5f90401c34c)

Example of static -0.5 Alpha value to illustrate:
![Godot_v4 2 2-stable_win64_pNnlYbJkAW](https://github.com/marcelb/GodotSSRWater/assets/308374/23c0201e-83d9-42c6-a7bc-f27b23c01c16)

Reply:
Good catch! will add it